test/e2e/e2e.go

```diff
@@ -0,0 +1,174 @@
+	MetallbOperatorDeploymentName = "metallboperator-controller-manager"
+	// MetallbOperatorDeploymentLabel contains the label of the MetalLB operator deployment
+	MetallbOperatorDeploymentLabel = "controller-manager"
+	// MetallbOperatorCRDName contains the name of the MetalLB operator CRD
+	MetallbOperatorCRDName = "metallbs.metallb.io"
+	// MetallbCRFile contains the Metallb custom resource deployment
+	MetallbCRFile = "metallb.yaml"
+	// MetallbDeploymentName contains the name of the MetalLB deployment
+	MetallbDeploymentName = "controller"
+	// MetallbDaemonsetName contains the name of the MetalLB daemonset
+	MetallbDaemonsetName = "speaker"
+)
+
+func RunE2ETests(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = Describe("validation", func() {
+	Context("MetalLB", func() {
+		It("should have the MetalLB operator deployment in running state", func() {
+			Eventually(func() bool {
+				deploy, err := testclient.Client.Deployments(MetallbNameSpace).Get(context.Background(), MetallbOperatorDeploymentName, metav1.GetOptions{})
+				if err != nil {
+					return false
+				}
+				return deploy.Status.ReadyReplicas == deploy.Status.Replicas
+			}, 5*time.Minute, 5*time.Second).Should(BeTrue())
+
+			pods, err := testclient.Client.Pods(MetallbNameSpace).List(context.Background(), metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("control-plane=%s", MetallbOperatorDeploymentLabel)})
+			Expect(err).ToNot(HaveOccurred())
+
+			deploy, err := testclient.Client.Deployments(MetallbNameSpace).Get(context.Background(), MetallbOperatorDeploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(pods.Items)).To(Equal(int(deploy.Status.Replicas)))
+
+			for _, pod := range pods.Items {
+				Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+			}
+		})
+
+		It("should have the MetalLB CRD available in the cluster", func() {
+			crd := &apiext.CustomResourceDefinition{}
+			err := testclient.Client.Get(context.Background(), goclient.ObjectKey{Name: MetallbOperatorCRDName}, crd)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("MetalLB deploy", func() {
+		var metallb *metallbv1alpha.Metallb
+		var metallbCRExisted bool
+
+		BeforeEach(func() {
+			metallb = &metallbv1alpha.Metallb{}
+			err := loadMetallbFromFile(metallb, MetallbCRFile)
+			Expect(err).ToNot(HaveOccurred())
+
+			metallbCRExisted = true
+			err = testclient.Client.Get(context.Background(), goclient.ObjectKey{Namespace: metallb.Namespace, Name: metallb.Name}, metallb)
+			if errors.IsNotFound(err) {
+				metallbCRExisted = false
+				Expect(testclient.Client.Create(context.Background(), metallb)).Should(Succeed())
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+
+		AfterEach(func() {
+			if !metallbCRExisted {
+				err := testclient.Client.Delete(context.Background(), metallb)
```

Comment:
This may cause status leak in between tests, because you are not sure the CRD is deleted when the test exits.
Better checking it's deleted inside an eventyually